Сетка страниц и крупных блоков на флексах

Artem-Sobolev91 opened the pull request:
Добрый день. Не много не понял формулировку задания, возможно что то лишнее сделал, а  где то пропустил. Посмотрите пожалуйста. 

---
:mortar_board: [Сетка страниц и крупных блоков на флексах](https://up.htmlacademy.ru/htmlcss-individual/1/user/2518753/tasks/7)

:boom: https://htmlacademy-htmlcss.github.io/2518753-sedona-1/5/

Artem-Sobolev91 committed:
Добавляет page-container

Artem-Sobolev91 committed:
Update catalog.html

Artem-Sobolev91 committed:
делает сетку навигации

Artem-Sobolev91 committed:
Делает сетку main

Artem-Sobolev91 committed:
Делает advantages

Artem-Sobolev91 committed:
Делает сетку services-item

Artem-Sobolev91 committed:
Добавляет flex соц сетям

Artem-Sobolev91 committed:
делает sticky-footer.

Не понял работает или нет. Ничего не меняется.

Artem-Sobolev91 committed:
Update styles.css

Artem-Sobolev91 committed:
Update styles.css

Artem-Sobolev91 committed:
Делает sticky footer

Не могу разобраться что должно выйти и как проверить. вроде нашел косяки, подправил в предыдущих комитах.

Artem-Sobolev91 committed:
Update styles.css

Artem-Sobolev91 committed:
исправляет опечатку

Artem-Sobolev91 committed:
Делает сетку шапки каталога

Artem-Sobolev91 committed:
Update catalog.html

Artem-Sobolev91 committed:
Добавляет контейнеры на каталог

main-container присвоил размер ,вместо отдельного дива. Так более правильно или отдельный див ,flex-grow: 1 не трогать?

Artem-Sobolev91 committed:
добавляет сетку к странице каталога.

Artem-Sobolev91 committed:
выравнивает product-list

Artem-Sobolev91 committed:
добавляет контейнер к секции фильтры

Artem-Sobolev91 committed:
добавляет сетку  "catalog-filter

Artem-Sobolev91 committed:
Редактирует.

Artem-Sobolev91 committed:
исправляет див

Artem-Sobolev91 committed:
Задает вопрос по "подписка"

1.Возможно стоит "subscription" опустить в footer?
2.Нужно ли весь боди обернуть в див и задать ему общие параметры ширины  и реализовать стики футер(не понял правда почему общий див.)

andreysgra reviewed (changes requested):
В данном проекте вся контентная область ограничена шириной 1200px и нет блоков в которых фоновый цвет тянется на всю ширину страницы, поэтому логичнее сделать общий контейнер для всего контента, например `div` с классом `.container` и в него поместить весь контент с хэдером и футером:
```
<div class="container">
  <header class="main-header"></header>

  <main class="main-container"></main>

  <footer class="main-footer"></footer>
</div>
```
В класс `.container` перенести свойства `width: 1200px; margin: 0 auto;`.

---

В этом проекте нет различий в стилизации тега `main` для главной страницы и внутренних страниц, соответственно нет необходимости использовать дополнительные классы `.main-inner` и `.main-index`, достаточно одного класса `.main-container`.

Artem-Sobolev91 committed:
Делает общий контейнер

Artem-Sobolev91 committed:
Добавляет общий контейнер

Artem-Sobolev91 committed:
Изменяет контейнер и его фоны.

Artem-Sobolev91 committed:
добавляет тень страницы

Artem-Sobolev91 committed:
удаляет space-between

Artem-Sobolev91 committed:
задает размеры с отступами adv-item

Artem-Sobolev91 committed:
добавляет дивы на img

все разъехалось. пришлось картинку в отдельный li убрать. задал item одинаковые размеры и отступы, уехала высота второго ряда. Возможно внутри там будет еще полоска и все выравним. Может это след тема, мне что то знаний не хватает .визуально вроде вырвнял.

Artem-Sobolev91 committed:
добавляет перенос.

Походу понял. Когда добавим линии  и отступы внутри элементов, мы потом размер отступов опять подкоректируем. Будем подгонять под макет. ЧТобы размер квадрата совпал под 400*385

Artem-Sobolev91 committed:
Удаляет див -футер.

Artem-Sobolev91 committed:
Update catalog.html

Artem-Sobolev91 committed:
Update styles.css

Artem-Sobolev91 committed:
Раскладываю карточки товаров.

Ширину не мог задать, когда поставил   max-width: 340px( было просто width-вылазило за границы);

Artem-Sobolev91 committed:
Выравниваю карточки.

Теперь ширина сжалась. Задал просто ширину

Artem-Sobolev91 committed:
редактрирует pagination

Artem-Sobolev91 committed:
Выстривает по 3 карточки.

box-sizing:border-box; задал. Все выстроилось.

Artem-Sobolev91 committed:
Задает бордер бокс для advantages-item

 размеры теперь совпадают.

Artem-Sobolev91 committed:
Update catalog.html

Artem-Sobolev91 commented:
В конечных комитах ,сам ответил на вопрос о размерах квадратов в advantages.

andreysgra commented on index.html:
```diff
@@ -8,131 +8,147 @@
+              <p>Седона — не аттракцион для туристов, там течёт своя жизнь</p>
+            </div>
+          </li>
+          <li>
+            <div class="advantages-item-img">
+              <img src="images/photo-1.jpg" width="800" height="385" alt="улица города">
+              <div>
+          </li>
```
Здесь разметку вы переделали, и не вполне корректно.

Было вот так:
```
       <li class="advantages-item advantages-item-wide">
          <h3>Настоящий городок</h3>
          <p>Седона — не аттракцион для туристов, там течёт своя жизнь</p>
          <img src="images/photo-1.jpg" width="800" height="385" alt="улица города">
        </li>
```
Изображение должно находиться в в одном элементе списка, что и описание, но не в разных.

Такая разметка будет правильной:
```
          <li class="advantages-item advantages-item-wide">
            <div class="advantages-item-text">
              <h3>Настоящий городок</h3>
              <p>Седона — не аттракцион для туристов, там течёт своя жизнь</p>
            </div>
            <div class="advantages-item-img">
              <img src="images/photo-1.jpg" width="800" height="385" alt="улица города">
              </div>
          </li>
```

Тоже самое нужно сделать и для другого элемента списка с изображением.

andreysgra reviewed (changes requested):
Есть ошибки валидации HTML обеих страниц.

Artem-Sobolev91 committed:
Убирает лишние li

Artem-Sobolev91 committed:
Переносит див после футер
